plugins/woocommerce/tests/php/src/Internal/DataStores/StockNotifications/StockNotificationsDataStoreTests.php

```diff
@@ -0,0 +1,535 @@
+		$this->assertEquals( null, $notification->get_date_modified() );
+		$this->assertEquals( null, $notification->get_date_subscribed() );
+		$this->assertEquals( null, $notification->get_date_notified() );
+
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->save();
+
+		$this->assertEquals( 1, $notification->get_id() );
+		$this->assertEquals( 1, $notification->get_product_id() );
+		$this->assertEquals( 1, $notification->get_user_id() );
+		$this->assertEquals( null, $notification->get_user_email() );
+		$this->assertEquals( 'pending', $notification->get_status() );
+		$this->assertFalse( $notification->is_queued() );
+		$this->assertEqualsWithDelta( 5, $notification->get_date_created()->getTimestamp(), time() );
+		$this->assertEqualsWithDelta( 5, $notification->get_date_modified()->getTimestamp(), time() );
+		$this->assertEquals( null, $notification->get_date_subscribed() );
+		$this->assertEquals( null, $notification->get_date_notified() );
+	}
+
+	/**
+	 * Test updating a notification with all properties.
+	 */
+	public function test_update_notification() {
+		// Create initial notification.
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->set_user_email( 'test@test.com' );
+		$notification->set_date_created( '2024-01-01 00:00:00' );
+		$notification->set_date_modified( '2024-01-01 00:00:00' );
+		$notification->save();
+
+		// Verify dates.
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_created()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_modified()->format( 'Y-m-d H:i:s' ) );
+
+		// Update all properties.
+		$notification->set_product_id( 2 );
+		$notification->set_user_id( 2 );
+		$notification->set_user_email( 'test2@test.com' );
+		$notification->set_status( 'active' );
+		$notification->set_date_subscribed( '2024-01-02 00:00:00' );
+		$notification->set_date_notified( '2024-01-02 00:00:00' );
+		$notification->set_is_queued( true );
+		$notification->save();
+
+		// Verify all properties were updated correctly.
+		$this->assertEquals( 2, $notification->get_product_id() );
+		$this->assertEquals( 2, $notification->get_user_id() );
+		$this->assertEquals( 'test2@test.com', $notification->get_user_email() );
+		$this->assertEquals( 'active', $notification->get_status() );
+		$this->assertEquals( '2024-01-02 00:00:00', $notification->get_date_subscribed()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '2024-01-02 00:00:00', $notification->get_date_notified()->format( 'Y-m-d H:i:s' ) );
+		$this->assertTrue( $notification->is_queued() );
+
+		// Verify modified date is updated.
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_created()->format( 'Y-m-d H:i:s' ) );
+		$this->assertNotEquals( '2024-01-01 00:00:00', $notification->get_date_modified()->format( 'Y-m-d H:i:s' ) );
```

Comment:
This is slightly confusing to me. I understand that in the first save, the date modified is retained because it was explicitly updated, but shouldn't it change to the time of the second save (on line 160)?

Reply:
Thanks, I somehow missed that. Then it makes sense.